spynnaker/pyNN/models/neuron/synapse_io.py

```diff
@@ -213,12 +213,16 @@ def get_max_row_info(
                 max_delayed_n_synapses)
 
         # Adjust for the allowed row lengths from the population table
-        undelayed_max_n_words = self._get_max_row_length(
-            undelayed_n_words, dynamics, population_table, in_edge,
-            max_undelayed_n_synapses)
-        delayed_max_n_words = self._get_max_row_length(
-            delayed_n_words, dynamics, population_table, in_edge,
-            max_delayed_n_synapses)
+        undelayed_max_n_words = 0
+        if undelayed_n_words > 0:
+            undelayed_max_n_words = self._get_max_row_length(
```

Comment:
I'm still not sure I get this.  Doesn't this function just return 0 when undelayed_max_n_words == 0?

Reply:
Yeah I'm not completely sure either; hence why I asked Peter for some help with it.

Reply:
when does that happen if structural plasticity keeps padding the row up to the required length?

Reply:
If you have a situation where all the delays require delay extensions, then undelayed_n_words =0 (and delayed_n_words != 0), or vice versa.  I'm wondering whether this is causing the problem... but as I said, I'm not completely sure.

Reply:
I'm testing it now by running a model with structural plasticity and setting --s_max 258 (this is an example). I want it to crash. If I recall correctly from when I wrote the initial issue, it would crash and it would just hang.

Reply:
OK, I have the issue now.  It does hang, because it is trying to reduce something down that will never reduce, because there is a "pad_to_length" set.  The trick would be to detect this specific case and error on it immediately.

Reply:
It probably needs either that function to be looked at in more detail, or else the function ```get_n_words_for_plastic_connections``` to be split up into a private version that ignores pad_to_length and therefore won't fall into the trap, and a public version that calls the private version after checking pad_to_length.

Reply:
(to be more specific, the function that would need to be looked at is ```get_max_synapses``` in ```SynapseDynamicsSTDP```)